src/frontend/src/lib/components/transactions/Transaction.svelte

```diff
@@ -27,7 +30,13 @@
 	<Card>
 		<span class="inline-block first-letter:capitalize"><slot /></span>
 
-		<RoundedIcon slot="icon" {icon} iconStyleClass={iconWithOpacity ? 'opacity-10' : ''} />
+		<div slot="icon">
+			{#if nonNullish(token)}
+				<TokenLogo data={token} badge={{ type: 'transaction', icon, ariaLabel: type }} />
+			{:else}
+				<RoundedIcon {icon} iconStyleClass={iconWithOpacity ? 'opacity-10' : ''} />
```

Comment:
`iconWithOpacity ? 'opacity-10' : ''` feels like a code duplicate, is that correct? In another PR can we try to avoid duplication if correct?

Reply:
for sure I can do in another PR, but I am not totally sure what you mean:

for example, pass `iconWithOpacity` as prop to `RoundedIcon`

Reply:
Done with:

- PR #3605